### DISCUSSION TITLE
Using "Save Row" from table's index screen?

### DISCUSSION THREAD

**Discussion by Grayda** [Budibase/budibase#2800] [Help]
I've got three tables, `Items`, `Templates` and `Colours`. `Items` and `Templates` contain the exact same columns: `Title` (text), and `Colours` (relationship to the `Colours` table, obviously). The idea of the `Templates` table is so that I can click a button on the Templates screen and copy a pre-made row to `Items` to save some time filling out the New Item form (which in reality has about a dozen different form elements).

But I've run into a few problems

 1. If I create a button with the onclick action of "Save Row", the row is never saved if I use Templates Table as the source, and if I use the Data Provider as the source, the row copied is blank.
 2. If I use the Data Provider as the source and manually add the fields like so: Title Column = `{{ Templates Table.Templates.Title }}` and Colours Column = `{{ Templates Table.Templates.Colours }}` then the `Title` column appears in the `Items` table, but the `Colours` relationship is empty

Even if I add the same button to the `/templates/:id` screen, the same thing happens, regardless of whether I use Form, Repeater, or Data Provider as the source. 

Am I missing something with the Save Row action?

**Comment by tobehn** [comment]
I had some success in a similar case today. 
You can store values from one table to another by referencing the new tables column to the origin tables field in the Save action.

![image](https://user-images.githubusercontent.com/40893107/135299897-507c326a-c6cc-42e4-9250-99bb779a6e8c.png)

What i do here:

on save i read the id of the row in MonLog table and write it into Parent field in MonLog-Replies table.
By this you should be able to write values from Template table to Items table.

I hope this helps.

**Reply by Grayda** [reply]
I tried this, but it didn't work. If I set up my example exactly like yours, No row is created. I have to change the data source to Data Provider and manually set the fields before I get anything, and I still don't get relationships filled.

**Reply by tobehn** [reply]
[ItemColorTemplates.txt](https://github.com/Budibase/budibase/files/7253996/ItemColorTemplates.txt)

Checkout this app.. Go to view a template and see the button in the bottom of the template.
Push and this template will be stored to Items.
I did not work with relations.

Hopefully i got you right.

**Reply by Grayda** [reply]
That works, but if I change the `colours` column to be a relationship, the relationships isn't set.

I found this discussion: https://github.com/Budibase/budibase/discussions/1705#discussioncomment-860755 and that told me two things:

 1. "You shouldn't need to declare the fields in your save record action. This should only be used if you want to send some custom value, that is not tied to a field by the same name", so if I just want to copy the row from one table to another, it should just copy without me having to specify the fields like you did with the `{{ Repeater.template.Title }}`
 2. Saving relationships to another table seems to be a bug?

**Reply by mjashanks** [reply]
Hey @Grayda - I just tried to put this together myself, and can see the limitation with relationships - it doesn't copy the relationship across.

This is actually related to this: https://github.com/Budibase/budibase/discussions/2617 - give it an upvote there, I will link this discussion.

I commented in there, checkout my video to make sure I've got you right!

**Reply by Grayda** [reply]
Yep @mjashanks, that's exactly what I'm doing, and what I'm experiencing.

My eventual goal is to have something like this:

![image](https://user-images.githubusercontent.com/4891563/135626128-bff6f4da-9d58-424f-b080-fef0a42d53b4.png)

So that you can just go to `/templates` and one-click to bulk-add items easily.

**Reply by gvlx** [reply]
Hi @Grayda,

I've written this guide: #3365
Hope it helps.

**Reply by Grayda** [reply]
> Hi @Grayda,
> 
> I've written this guide: #3365 Hope it helps.

I'm not quite sure how to use your guide, because this seems to be for text-based items (e.g. adding notes to an item), instead of for setting relationships (adding a relationship to an item)
